components/user-settings/UserSettings.tsx

```diff
@@ -0,0 +1,124 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import {
+  Button,
+  EButtonTypes,
+  EIConTypes,
+  EInputFieldGroupType,
+  EStateInputFieldGroup,
+  ETypographyLevels,
+  Heading,
+  InputFieldGroup,
+} from '@ost-cas-fee-adv-23-24/elbmum-design';
+import useUserInfo from '@/hooks/useUserInfo';
+
+export interface INewUserSettings {
+  firstName: string;
+  lastName: string;
+  email: string;
+}
+
+interface IProps {
+  onClose: () => void;
+  onSave: (options: INewUserSettings) => void;
+}
+const UserSettings = ({ onClose, onSave }: IProps) => {
+  const { firstName, lastName } = useUserInfo();
+  const [firstNameIntern, setFirstNameIntern] = useState('');
+  const [lastNameIntern, setLastNameIntern] = useState('');
+  const [emailIntern, setEmailIntern] = useState('');
+
+  useEffect(() => {
+    setFirstNameIntern(firstName);
+    setLastNameIntern(lastName);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why do you `eslint-disable-next-line` for `exhaustive-deps`?

Reply:
yes, otherwise an alert will appear to tell you to add dependencies. I just want to run that once. that's why no dependency is needed.